Controller/Export/Category.php

```diff
@@ -0,0 +1,99 @@
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2020 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Tagging\Controller\Export;
+
+use Magento\Framework\App\Action\Context;
+use Magento\Store\Model\Store;
+use Nosto\Model\AbstractCollection;
+use Nosto\Model\Category\CategoryCollection;
+use Nosto\NostoException;
+use Nosto\Tagging\Helper\Account as NostoHelperAccount;
+use Nosto\Tagging\Helper\Scope as NostoHelperScope;
+use Nosto\Tagging\Model\Category\CollectionBuilder;
+
+/**
+ * Category export controller used to export category tree to Nosto.
+ * This controller will be called by Nosto when a new account has been created
+ * from the Magento backend. The controller is public, but the information is
+ * encrypted with AES, and only Nosto can decrypt it.
+ */
+class Category extends Base
+{
+    public const PARAM_PREVIEW = 'preview';
+
+    /** @var CollectionBuilder  */
+    private CollectionBuilder $nostoCollectionBuilder;
+
+    /**
+     * @param Context $context
+     * @param NostoHelperScope $nostoHelperScope
+     * @param NostoHelperAccount $nostoHelperAccount
+     * @param CollectionBuilder $collectionBuilder
+     */
+    public function __construct(
+        Context $context,
+        NostoHelperScope $nostoHelperScope,
+        NostoHelperAccount $nostoHelperAccount,
+        CollectionBuilder $collectionBuilder
+    ) {
+        parent::__construct($context, $nostoHelperScope, $nostoHelperAccount);
+        $this->nostoCollectionBuilder = $collectionBuilder;
+    }
+
+    /**
+     * @param Store $store
+     * @param int $limit
+     * @param int $offset
+     * @return AbstractCollection|CategoryCollection
+     * @throws NostoException
+     */
+    public function buildExportCollection(Store $store, int $limit = 100, int $offset = 0)
```

Comment:
Can be called with https://magento245.dev.nos.to:8443/nosto/export/category?&limit=2&offset=1&___store=default